tests/scf-cholesky-basis/input.dat

```diff
@@ -15,16 +15,17 @@ e_canonical   = -230.64152543331906
 # indices. Since there's no guarantee that the same functions get
 # picked every time, we need to allow for changes in the energy that
 # go to zero when the tolerance is tightened.
+# MKL 2020.1 seems to often pick different fns, so need to relax the tolerance from 1.e-8 to 2.e-8
 
 tol_4 = 3
 tol_5 = 4
 tol_6 = 5
-tol_7 = 7
-tol_8 = 7
-tol_9 = 7
-tol_10 = 8
-tol_11 = 8
-tol_can = 8
+tol_7 = 2.e-8
+tol_8 = 2.e-8
+tol_9 = 2.e-8
+tol_10 = 2.e-8
+tol_11 = 2.e-8
+tol_can = 2.e-8
```

Comment:
I'm not sure you should change the canonical threshold

Reply:
good point

Reply:
nope, canonical, too: `Energy with canonical orthogonalization: computed value                  !
! (-230.6415254219) does not match (-230.6415254333) to atol=1e-08 by difference   !
! (0.0000000114). ` 
does this change the analysis of the situation, @susilehtola?

Reply:
@loriab then this is not related to Cholesky but mkl changing linear algebra...

Reply:
The canonical calc is dependent on the Cholesky calcs through `guess read`. I haven't been able to get the canonical calc to produce the reference value when run alone (off by about 0.001857 [E_h]) with 2019.4 or 2020.1. Add back the Cholesky 1e-4 and then you get the canonical passing for 2019.4 for the 1.1e-8 error for 2020.1 .

```
<     One-Electron Energy =                -715.6736276673161683
<     Two-Electron Energy =                 281.5139810571366183
<     Total Energy =                       -230.6415254410109696
---
>     One-Electron Energy =                -715.6736277971549498
>     Two-Electron Energy =                 281.5139811795980904
>     Total Energy =                       -230.6415254483882791
```

But overall, MKL values are changing results. Here's two other basis sets besides the native one in the test (fitting basis clearly not appropriate for datz). They show differences, but the test basis is by far the largest magnitude.

```
basis   mkl     DF-UHF                  Norb    Naux
-----   ---     ------                  ----    ----
 adz    2019.4  -230.6918 9098 9512     192     708
 adz    2020.1  -230.6918 9098 845158

test    2019.4  -230.6415 2544 83882791 270     558
test    2020.1  -230.6415 2544 10109696

datz    2019.4  -230.7736 4428 528847   564     558
datz    2020.1  -230.7736 4428 529125
```

Here's the [RN for MKL](https://software.intel.com/content/www/us/en/develop/articles/intel-math-kernel-library-release-notes-and-new-features.html). Not seeing anything likely. Psi should be more reliable than this (all with d_conv=10), so I'm thinking we'll need to ban 2020.1. Glad to hear views.

Reply:
Can you get the same value from 2019 and 2020 from a basic canonical calc?

Reply:
> 
> Here's the [RN for MKL](https://software.intel.com/content/www/us/en/develop/articles/intel-math-kernel-library-release-notes-and-new-features.html). Not seeing anything likely. Psi should be more reliable than this (all with d_conv=10), so I'm thinking we'll need to ban 2020.1. Glad to hear views.

My 2 cents is that banning specific versions of specific linalg packages, with no other reason than "it changes results", is something that makes me feel uneasy. It feels like a bandaid over a bigger problem. For example what if all future versions of MKL start behaving like 2020.1?

With that said, I do like consistent results. Personally, a few weeks ago I spent about an entire day trying to make sense of why a certain Molpro calculation fails to converge for someone else, while the same exact input works for me.
As it turns out, the calculation only converges with an older version of Molpro, and only with an AMD CPU. No, I am serious, for some unknown reason (probably related to CPU dispatching), that input only converges on AMD CPUs running a given older version.

Reply:
problem hits for charged/uhf (haven't disentagled) systems between pk/tz and df/atz. Neutral singlet water shows no problems matching to 1e-10 for any reference, any mkl, any scf_type.

```
memory 3 gb

molecule {
-1 2
C          0.00000        0.00000        1.39495
C          0.00000        1.20806        0.69748
C          0.00000        1.20806       -0.69748
C          0.00000        0.00000       -1.39495
C          0.00000       -1.20806       -0.69748
C          0.00000       -1.20806        0.69748
H          0.00000        0.00000        2.47753
H          0.00000        2.14560        1.23877
H          0.00000        2.14560       -1.23877
H          0.00000        0.00000       -2.47753
H          0.00000       -2.14560       -1.23877
H          0.00000       -2.14560        1.23877
}

#set basis aug-cc-pVTZ
set basis cc-pVTZ
#set basis cc-pVQZ
#set basis aug-cc-pVQZ
set scf_type pk
#set scf_type df
set d_convergence 10
set e_convergence 11
set df_scf_guess false
set guess core
#set reference rohf
set reference uhf
e = energy('scf')
print(e)

pk_qz_2019  = -76.06462080005818
pk_aqz_2019 = -76.06578901994756
df_qz_2019 =  -76.06462015458904
df_aqz_2019 = -76.06578796456354

pk_tz_rohf_2019 = -230.674287472723
pk_tz_uhf_2019 = -230.684552455135      # passes 2019.4, passes 2020.1
df_atz_uhf_2019 = -230.6966011258728    # passes 2019.4, fails 2020.1
ref = pk_tz_uhf_2019
compare_values(ref, variable('SCF TOTAL ENERGY'), 10, 'SCF energy')  #TEST
```

Reply:
I'm echoing, "[B]anning specific versions of specific linalg packages, with no other reason than "it changes results", is something that makes me feel uneasy. It feels like a bandaid over a bigger problem. For example what if all future versions of MKL start behaving like 2020.1?"